### ISSUE TITLE
Set model to eval mode for better performance

### ISSUE THREAD

**Issue by finetuneanon** [KoboldAI/KoboldAI-Client#11]
From some reports, people got lower performance from my finetuned models, but copying over the original model's config seems to fix it. The only significant difference is that gradient checkpointing is enabled for my models, which should only cause a difference when training. However, it seems that KoboldAI doesn't set the model to evaluation mode. Adding the following on line 240 in aiserver.py should fix it: model = model.eval()

From what I see in the transformers code, adding the device argument to pipeline should not actually do anything when the model is already instantiated and passed in directly. In that case, calling model.cuda(0) instead should work to load it on the GPU.

**Comment by KoboldAI** [comment]
Strangely, when I was first implementing transformers, I couldn't actually get it to use my GPU **unless** I added that device assignment. More recently I tried replacing the transformers pipeline with the torch.load implementation from your colab notebook, but I lost the ability to run the 2.7B models on my 8GB card (CUDA would OOM). I could avoid the OOM by clearing the checkpoint keys before assigning it to the model, but the generator results became just a random assortment of nonsense, so I scrapped the experiment.

**Comment by KoboldAI** [comment]
I'm not seeing a noticeable difference using either eval() or cuda(0). I have the following lines of code:
`if(vars.hascuda and vars.usegpu):
    model = model.cuda(0).eval()
    generator = pipeline('text-generation', model=model, tokenizer=tokenizer, device=0)`
I'm still seeing ~18 second generation times, and the GPU is definitely being used as I can see the dedicated memory usage sitting around 6-7Gb in Task Manager. I actually don't know when this started happening, I definitely remember having 3-5 second generation times at one point.

**Comment by finetuneanon** [comment]
Maybe that is not the issue then. Is that with the original EleutherAI model?

**Comment by KoboldAI** [comment]
Ran all three just now.
The stock 1.3B and 2.7B models are giving me ~3-5s generations without the .cuda(0).eval() additions. 
Neo-Horni takes ~18s regardless of whether .cuda(0).eval() are in use.

**Comment by finetuneanon** [comment]
Interesting, thanks for testing. Does it get faster if you change gradient_checkpointing to false in its config file?

**Comment by KoboldAI** [comment]
No, no change with gradient_checkpointing. However, if I swap Horni's config file for the one from stock 2.7B, generation times drop to ~6-7s

**Comment by finetuneanon** [comment]
Didn't really see any other differences between the files. Very strange.

**Comment by finetuneanon** [comment]
Actually, could you please try once more with "use_cache" set to true in horni's config?

**Comment by KoboldAI** [comment]
Oh yeah, that did it. Lightning quick responses with use_cache set to true.

**Comment by finetuneanon** [comment]
In that case, adding use_cache=True to the generator() call should do the trick.

**Comment by KoboldAI** [comment]
Added use_cache=True to generate calls in tonight's push. Works much better on my end, let me know if reports of sluggishness continue. Thanks again.
